Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -85,39 +101,57 @@ namespace ROS2
 
     void ROS2SpawnerComponent::SpawnEntity(const SpawnEntityRequest request, SpawnEntityResponse response)
     {
-        AZStd::string_view key(request->name.c_str(), request->name.size());
+        AZStd::string_view spawnable_name(request->name.c_str(), request->name.size());
+        AZStd::string_view spawn_point_name(request->xml.c_str(), request->xml.size());
 
-        if (!m_spawnables.contains(key))
+        auto spawn_points = GetSpawnPoints();
+
+        if (!m_spawnables.contains(spawnable_name))
         {
             response->success = false;
             response->status_message = "Requested spawnable name not found";
             return;
         }
 
-        if (!m_tickets.contains(key))
+        if (!m_tickets.contains(spawnable_name))
         {
             // if a ticket for this spawnable was not created but the spawnable name is correct, create the ticket and then use it to
             // spawn an entity
-            auto spawnable = m_spawnables.find(key);
+            auto spawnable = m_spawnables.find(spawnable_name);
             m_tickets.emplace(spawnable->first, AzFramework::EntitySpawnTicket(spawnable->second));
         }
 
         auto spawner = AZ::Interface<AzFramework::SpawnableEntitiesDefinition>::Get();
 
         AzFramework::SpawnAllEntitiesOptionalArgs optionalArgs;
 
-        optionalArgs.m_preInsertionCallback = [this, position = request->initial_pose](auto id, auto view)
+        AZ::Transform transform;
+
+        if (spawn_points.contains(spawn_point_name))
         {
-            this->PreSpawn(
-                id,
-                view,
-                AZ::Transform(
-                    AZ::Vector3(position.position.x, position.position.y, position.position.z),
-                    AZ::Quaternion(position.orientation.x, position.orientation.y, position.orientation.z, position.orientation.w),
-                    1.0f));
+            transform = spawn_points.at(spawn_point_name).pose;
+        }
+        else if (spawn_point_name == "default")
```

Comment:
we could also simplify this by treating the default spawn point just like any other, under the key "default" in spawn_points (this could be implicit). Getter would then always include at least the default spawn point.